amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/declaration/Oas2SecuritySettingsParser.scala

```diff
@@ -1,37 +1,53 @@
 package amf.plugins.document.webapi.parser.spec.declaration
 
 import amf.core.annotations.VirtualObject
+import amf.core.parser.{Annotations, ScalarNode, YMapOps}
+import amf.core.utils.{AmfStrings, Lazy}
 import amf.plugins.document.webapi.contexts.parser.oas.OasWebApiContext
 import amf.plugins.document.webapi.parser.spec.common.AnnotationParser
-import org.yaml.model.{YMap, YMapEntry}
-import amf.plugins.domain.webapi.metamodel.security.{
-  OAuth2SettingsModel,
-  HttpSettingsModel,
-  OAuth2FlowModel,
-  OpenIdConnectSettingsModel
-}
+import amf.plugins.domain.webapi.metamodel.security.{OAuth2FlowModel, OAuth2SettingsModel}
 import amf.plugins.domain.webapi.models.security._
-import amf.core.parser.{Annotations, ScalarNode, YMapOps}
-import amf.core.utils.{Lazy, AmfStrings}
+import amf.validations.ParserSideValidations
+import org.yaml.model.YMap
 
 class Oas2SecuritySettingsParser(map: YMap, scheme: SecurityScheme)(implicit ctx: OasWebApiContext)
     extends OasLikeSecuritySettingsParser(map, scheme) {
 
   override def parse(): Option[Settings] = {
-    produceSettings.map { settings =>
-      val parsedSettings = settings match {
-        case s: OAuth1Settings => parseOauth1Settings(s)
-        case s: OAuth2Settings => parseOauth2Settings(s)
-        case s: ApiKeySettings => parseApiKeySettings(s)
-        case defaultSettings =>
-          map
-            .key("settings".asOasExtension)
-            .map(entry => parseDynamicSettings(entry.value.as[YMap], defaultSettings))
-            .getOrElse(defaultSettings)
+    produceSettings
+      .map { settings =>
+        val parsedSettings = settings match {
+          case s: OAuth1Settings => parseOauth1Settings(s)
+          case s: OAuth2Settings => parseOauth2Settings(s)
+          case s: ApiKeySettings => parseApiKeySettings(s)
+          case defaultSettings =>
+            map
+              .key("settings".asOasExtension)
+              .map(entry => parseDynamicSettings(entry.value.as[YMap], defaultSettings))
+              .getOrElse(defaultSettings)
+        }
+        parsedSettings.annotations += VirtualObject()
+        parseAnnotations(parsedSettings)
       }
-      parsedSettings.annotations += VirtualObject()
-      parseAnnotations(parsedSettings)
-    }
+      .orElse({ handleNoSettings() })
+  }
+
+  private def handleNoSettings(): Option[Settings] = {
+    val schemeType = scheme.`type`.value()
+
+    def settingsPresent: Boolean =
+      map.entries.exists(element => {
+        val field = element.key.as[String]
+        field != "description" && field != "name"
+      })
+
+    if (schemeType == "Basic Authentication" && settingsPresent)
```

Comment:
```suggestion
   const BASIC_AUTHENTICATION_SCHEME = "Basic Authentication"
    if (schemeType == BASIC_AUTHENTICATION_SCHEME && settingsPresent)
```
Extract constant to a more significant place